tests/problem2Tests.rb

```diff
@@ -0,0 +1,53 @@
+require 'test/unit'
+require '../src/problem2'
+
+class Problem2Tests < Test::Unit::TestCase
+
+  def setup
+    @problem2 = Problem2.new
+    @testOddSizeData = [1,9,2,8,3,7,4,6,5]
+    @testEvenSizeData = [10,1,9,2,8,3,7,4,6,5]
+    @testOrderedData = (1..100).to_a
+    @testRandomData = [512,100,2,1,30,140,10,170,80,0]
+    @testEqualMaxValueData = [1111,7,2,1,30,140,10,170,80,1111]
+    @testDataEmptyArray = []
+    @testDataHasOneElement = [9]
+  end
+
+  def teardown
+
+  end
+
+  def test_inputOddSizeHasExpectedResult
+    assert_equal([9,8],@problem2.calculateMaximum(@testOddSizeData))
+  end
+
+  def test_inputEvenSizeHasExpectedResult
+    assert_equal([10,9],@problem2.calculateMaximum(@testEvenSizeData))
+  end
+
+  def test_equalMaxValueHasExpectedResult
+    assert_equal([1111,1111],@problem2.calculateMaximum(@testEqualMaxValueData))
+  end
+
+  def test_emptyArrayHasExpectedResult
+    assert_equal([nil,nil],@problem2.calculateMaximum(@testDataEmptyArray))
+  end
+
+  def test_oneElementArrayHasExpectedResult
+    assert_equal([9,9],@problem2.calculateMaximum(@testDataHasOneElement))
```

Comment:
Don't believe that this is correct. You only have 1 number in the array.

Reply:
In the previous implementation, I was returning [9,null].
Then I read this comment:

(red) Add test for array with one element
(green) Return both maximus as the only element in the array

Reply:
:+1: Talked with @alexandru-calinoiu and any solution is ok since the problem statement can be interpreted in multiple ways. No change required here.